### PR TITLE
make getAgent public

### PR DESCRIPTION
**Pull request by ericsciple**
This PR makes getAgent a public function. This function is useful when you need an http agent that handles routing through a proxy server - depending upon the url and proxy environment variables.

For example, when instantiating the Octokit REST API client.

### CODE REVIEW

File: __tests__/proxy.test.ts [hunk 1]
```diff
@@ -1,89 +1,130 @@
...
+        // Stop proxy server
+        await new Promise((resolve) => {
+            _proxyServer.once('close', () => resolve())
+            _proxyServer.close()
+        })
     })
-  
+
     it('does not return proxyUrl if variables not set', () => {
-        _clearVars();
```
Comment (ericsciple) [inline __tests__/proxy.test.ts]: _clearVars moved to beforeEach

File: package.json [hunk 2]
```diff
@@ -23,10 +23,13 @@
   "homepage": "https://github.com/actions/http-client#readme",
   "devDependencies": {
     "@types/jest": "^24.0.25",
-    "@types/node": "^13.1.5",
+    "@types/node": "^12.12.24",
```
Comment (ericsciple) [inline package.json]: 12 is more accurate

File: package.json [hunk 3]
```diff
@@ -23,10 +23,13 @@
...
     "jest": "^24.9.0",
-    "nock": "^11.7.2",
+    "proxy": "^1.0.1",
```
Comment (ericsciple) [inline package.json]: proxy is used in the unit test to stand up a proxy server

**Comment by bryanmacfarlane** [comment]
LGTM

**Comment by bryanmacfarlane** [comment]
is it possible to add a test with a no_proxy setting, make the request and confirm it doesn't connect through the proxy?

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@actions/http-client",
-  "version": "1.0.1",
+  "version": "1.1.0",
```
Comment (bryanmacfarlane) [inline package.json]: I would just bump the patch version.  we want folks to get these changes and it's part of rounding out what 1.0.x offered

File: __tests__/proxy.test.ts [hunk 5]
```diff
@@ -1,89 +1,154 @@
...
+    it('does basic https get request when bypass proxy', async () => {
+        process.env['https_proxy'] = _proxyUrl
+        process.env['no_proxy'] = 'httpbin.org'
+        const httpClient = new httpm.HttpClient();
+        let res: httpm.HttpClientResponse = await httpClient.get('https://httpbin.org/get');
+        expect(res.message.statusCode).toBe(200);
+        let body: string = await res.readBody();
+        let obj: any = JSON.parse(body);
+        expect(obj.url).toBe("https://httpbin.org/get");
+        expect(_proxyConnects).toHaveLength(0)
```
Comment (bryanmacfarlane) [inline __tests__/proxy.test.ts]: cool

**Comment by bryanmacfarlane** [comment]
Looked through the changes and I think this is ready to merge.

File: package.json [hunk 6]
```diff
@@ -23,10 +23,13 @@
...
-    "@types/node": "^13.1.5",
+    "@types/node": "^12.12.24",
     "jest": "^24.9.0",
-    "nock": "^11.7.2",
+    "proxy": "^1.0.1",
     "ts-jest": "^24.3.0",
     "typescript": "^3.7.4"
+  },
+  "dependencies": {
+    "tunnel": "0.0.6"
```
Comment (joshmgross) [inline package.json]: @ericsciple I don't see where `tunnel` is used. Should this be moved to `devDependencies` or removed?

**Pull request merged**